albums/views.py

```diff
@@ -1,28 +1,53 @@
-from django.shortcuts import render
-from django.http import HttpResponse
-from .forms import albumForm
-from django.views import View
+from rest_framework import generics, mixins, permissions
+from rest_framework.response import Response
+from rest_framework.pagination import LimitOffsetPagination
+from rest_framework.views import APIView
+from django.forms import ValidationError
+from .serializers import AlbumSerializer
+from .filters import AlbumsFilters
+from .models import Albums
 
-class albums(View):
-    def get(self, request):
-        return HttpResponse('Welcome From Albums Page.')
-
-class createAlbum(View):
-    form = albumForm
-    template = 'createAlbum.html'
+class allAlbums(generics.ListAPIView, mixins.ListModelMixin, mixins.CreateModelMixin):
+    queryset = Albums.objects.filter(approved=True)
+    serializer_class = AlbumSerializer
+    permission_classes = [permissions.IsAuthenticatedOrReadOnly]
+    pagination_class = LimitOffsetPagination
+    filterset_class = AlbumsFilters
 
     def get(self, request):
-        if request.user.is_authenticated:
-            newAlbum = self.form()
-            return render(request, self.template, {'albumForm': newAlbum})
-        else:
-            return HttpResponse('Unauthenticated user, please login to access this page.')
+        return self.list(request) 
 
     def post(self, request):
-        if request.user.is_authenticated:
-            newAlbum = self.form(request.POST)
-            if newAlbum.is_valid():
-                newAlbum.save()
-            return render(request, self.template, {'albumForm': newAlbum})
+        if not hasattr(request.user, 'artist'):
```

Comment:
it is better to add a permission class to check if the user is an artist or not. first, create the permissions.py:
```python
from rest_framework import permissions

from artists.models import Artist


class IsArtistPermission(permissions.BasePermission):
    def has_permission(self, request, view):
        if request.method in permissions.SAFE_METHODS:
            return True
        return Artist.objects.filter(user_id = request.user.id).exists()
```

and add that permission to permission_class